tests/sam_learning_test.py

```diff
@@ -426,3 +426,77 @@ def test_learn_action_model_returns_learned_model(elevators_sam_learning: SAMLea
     learned_model, learning_report = elevators_sam_learning.learn_action_model([elevators_observation])
     print(learning_report)
     print(learned_model.to_pddl())
+
+
+def test_learn_action_model_with_hard_policy_deletes_negative_preconditions(
```

Comment:
since the tests are the same as in MA-SAM apply the changes from there to here as well.